src/shogun/regression/svr/LibLinearRegression.cpp

```diff
@@ -71,26 +69,16 @@ LibLinearRegression::~LibLinearRegression()
 {
 }
 
-bool LibLinearRegression::train_machine(std::shared_ptr<Features> data)
+bool LibLinearRegression::train_machine(
+    const std::shared_ptr<Features>& data, const std::shared_ptr<Labels>& labs)
 {
 
-	if (data)
-		set_features(data->as<DotFeatures>());
-
-	ASSERT(features)
-	ASSERT(m_labels && m_labels->get_label_type()==LT_REGRESSION)
+	const auto features = data->as<DotFeatures>();
+	auto labels = labs->as<RegressionLabels>();
 
-	auto num_train_labels=m_labels->get_num_labels();
 	auto num_feat=features->get_dim_feature_space();
 	auto num_vec=features->get_num_vectors();
 
-	if (num_vec!=num_train_labels)
-	{
-		error("number of vectors {} does not match "
-				"number of training labels {}",
-				num_vec, num_train_labels);
-	}
```

Comment:
shouldn't this check stay? but use `labels->get_num_labels()` instead?

Reply:
I plan to add this check to `train_machin` of `Machine`, then we don't have to write this check for every class.

Reply:
ah yes, I saw that it is already there, cool!